### DISCUSSION TITLE
Cannot find name 'PromiseSettledResult'

### DISCUSSION THREAD

**Discussion by valekar** [supabase/supabase#4941] [Questions] [answered]
When try to run my typescript project, I am getting ```Cannot find name 'PromiseSettledResult'``` error. Any idea on how to resolve this?  

Edit1 : The supbase-js version that I am using is **1.29.1**

**Comment by tylerscott7** [comment, marked as answer]
Unsure which version this starts to occur in, but I downgraded from 1.29.1->1.28.5 w/ npm install @supabase/supabase-js@1.28.5

**Reply by valekar** [reply]
Thanks that did the trick, I also downgraded it from 1.29.1 to the version that you have mentioned above. I appreciate your help. Send me your solana USDC token address, will buy you a cup of coffee. :))

**Comment by gthau** [comment]
Hi, the issue is not to downgrade the lib version.
It has to do with the target ES version. In tsconfig.json of your project, there is:
- a key "target" which defines to which Javascript version Typescript will transpile to
- a key "lib" to include polyfills for features missing in your target Javascript version

In your case, the lib wants to use `Promise.allSettled` (and the associated type `PromiseSettledResult`) which were added in ES2020. So if your "target" Javascript version is lower than ES2020, those don't exist therefore Typescript fails.

To fix this you have 2 options:
1. set "target" to "ES2020"
```json
{
  "compilerOptions": {
    // ...
    "target": "es2020"
}
```
2. keep "target" to the current value you have (es5, es6, es2017, es2018, es2019) but add the missing feature in "lib" (in this case you can either add the polyfills for the whole es2020 spec by using "es2020" or add only the polyfill for the Promise part of the es2020 spec by adding "ES2020.Promise")
```json
{
  "compilerOptions": {
    // ...
    "target": "es2019",
    "lib": [
       "ES2020.Promise"
    ]
}
```

**Reply by BranchManager69** [reply]
THANK YOU SIR!!!!!
